src/decode.c

```diff
@@ -250,6 +259,16 @@ int process_map(msgpack_object_map *map, schedule_event_t **t)
     return ret_val;
 }
 
+int decode_string_type (msgpack_object *key, msgpack_object *val, schedule_t **t)
+{
+    (void ) key;
+
+    memset((*t)->tz_struct.tz, 0, MAX_TIME_ZONE_SIZE);
+    strncpy((*t)->tz_struct.tz, val->via.str.ptr, val->via.str.size);
```

Comment:
Since we can't validate (due to various complexities), we should just `strdup()` into a `char*`